manifest: update hal_nordic revision

ankuns opened the pull request:
This commit updates revision of hal_nordic to bring the latest changes in the nRF IEEE 802.15.4 driver.

ankuns committed:
manifest: update hal_nordic revision

This commit updates revision of hal_nordic to bring the latest changes
in the nRF IEEE 802.15.4 driver.

Signed-off-by: Andrzej Kuroś <andrzej.kuros@nordicsemi.no>

anangl commented on west.yml:
```diff
@@ -193,7 +193,7 @@ manifest:
         - hal
     - name: hal_nordic
-      revision: fae15426c6b5a1f67362d508cf51b691ae5ab4b4
+      revision: pull/270/head
```
Update with SHA: `c563b0e56dfa7e3915637b12b3428e6de36012be`.

ankuns committed:
modules: hal_nordic: remove setting of NRF_802154_USE_RAW_API

The non-raw API has been removed from the nRF 802.15.4 Radio Driver.
There is no need to set NRF_802154_USE_RAW_API macro anymore in
CMakeLists.txt that integrates the driver.

Signed-off-by: Andrzej Kuroś <andrzej.kuros@nordicsemi.no>
